### PR TITLE
Safronov mikhail. Lab 1. Var 1.

### PR DESCRIPTION
**Pull request by yourd8ne**

### CODE REVIEW

File: clang/test/lab1/safronov_mikhail/test.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        //CHECK-NEXT: |_var
+        float var;
+    };
+};
+
+// CHECK: TemplateClass
+template<typename T> class TemplateClass {
+    //CHECK-NEXT: Tvariable
+    T Tvariable;
+};
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test.cpp:48]: Please add tests for static member field and template instantiation class.

File: clang/test/lab1/safronov_mikhail/test.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes %s 1>&1 | FileCheck %s
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test.cpp]:
```suggestion
// RUN: %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes %s 2>&1 | FileCheck %s
```

File: clang/labs/lab1/safronov_mikhail/PrintClasses.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  std::unique_ptr<clang::ASTConsumer>
+  CreateASTConsumer(clang::CompilerInstance &Compiler,
+                    llvm::StringRef InFile) override {
+    return std::make_unique<PrintClassesConsumer>();
+  }
+
+protected:
+  bool ParseArgs(const clang::CompilerInstance &Compiler,
+                 const std::vector<std::string> &args) override {
+    return true;
```
Comment (m-ly4) [inline clang/labs/lab1/safronov_mikhail/PrintClasses.cpp:54]: Please add `help` argument and test for it.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

**Comment by yourd8ne** [comment]
complete

**Comment by m-ly4** [comment]
> complete

For static members chack thank you. What about `help` argument?

**Comment by yourd8ne** [comment]
my bad. add help arg and test it complete.

File: clang/test/lab1/safronov_mikhail/test.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes --help %s 2>&1 | FileCheck %s
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test.cpp]:
```suggestion
// RUN: not %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes --help %s 2>&1 | FileCheck %s
```

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/test/lab1/safronov_mikhail/test.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
+// RUN: not %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes --help %s 2>&1 | FileCheck %s
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test.cpp]: It would be nice to split test for main fuctionality and help argument check.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/test/lab1/safronov_mikhail/test.cpp (hunk 5 above)
Comment (yourd8ne) [reply clang/test/lab1/safronov_mikhail/test.cpp]: it is not working:'(

File: tr.txt [hunk 6]
```diff
@@ -0,0 +1,10079 @@
+ocaml_make_directory: phony
```
Comment (m-ly4) [inline tr.txt]: For what this file is added?

File: clang/test/lab1/safronov_mikhail/test_help_args.cpp [hunk 7]
```diff
@@ -0,0 +1 @@
+// RUN: not %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes --help %s 2>&1 | FileCheck %s
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test_help_args.cpp]: Here should be at least one CHECK line.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: tr.txt (hunk 6 above)
Comment (yourd8ne) [reply tr.txt]: check targets

File: clang/test/lab1/safronov_mikhail/test_help_args.cpp (hunk 7 above)
Comment (yourd8ne) [reply clang/test/lab1/safronov_mikhail/test_help_args.cpp]: I added one check from the last test, it doesn’t work again

File: clang/test/lab1/safronov_mikhail/test.cpp [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+template<typename T> class TemplateClass2 {
+    //CHECK-NEXT: |_Tvariable1
+    T Tvariable1;
+    //CHECK-NEXT: |_Tvariable2
+    static T Tvariable2;
+    //CHECK-NEXT: |_TemplateClass2<T>
+    TemplateClass2() {};
+};
+
+// RUN: %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes --help %s 1>&1 | FileCheck %s --check-prefix=HELP
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test.cpp]:
```suggestion
// RUN: %clang_cc1 -load %llvmshlibdir/PrintClassesPlugin%pluginext -plugin print-classes -plugin-arg-print-classes --help %s 2>&1 | FileCheck %s --check-prefix=HELP
```

File: clang/test/lab1/safronov_mikhail/test.cpp [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+// HELP: Text
```
Comment (m-ly4) [inline clang/test/lab1/safronov_mikhail/test.cpp]:
```suggestion
// HELP: Text
// HELP-NOT: |_
```

File: clang/test/lab1/safronov_mikhail/test.cpp (hunk 8 above)
Comment (yourd8ne) [reply clang/test/lab1/safronov_mikhail/test.cpp]: complete

File: clang/test/lab1/safronov_mikhail/test.cpp (hunk 9 above)
Comment (yourd8ne) [reply clang/test/lab1/safronov_mikhail/test.cpp]: complete

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**